craft_parts/plugins/ant_plugin.py

```diff
@@ -0,0 +1,183 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2022 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The Ant plugin."""
+
+import logging
+from typing import Any, Dict, List, Optional, Set, cast
+
+from craft_parts import errors
+
+from . import validator
+from .base import Plugin, PluginModel, extract_plugin_properties
+from .properties import PluginProperties
+
+logger = logging.getLogger(__name__)
+
+
+class AntPluginProperties(PluginProperties, PluginModel):
+    """The part properties used by the Ant plugin."""
+
+    ant_build_targets: List[str] = []
+    ant_build_file: Optional[str] = None
```

Comment:
in the core18 plugin this as called `ant-buildfile`, because the Ant option is called `-buildfile`, but I thought I'd keep it consistent with `ant-build-targets`.